controllers/discovery/searchUser.js

```diff
@@ -18,15 +18,32 @@ const SearchUser = async (req, res) => {
     });
 
   try {
+    let user_topics = await sequelize.query(
+      `select tp.topic_id from topics as tp
+      left join user_topics as utp on tp.topic_id = utp.topic_id
+      where utp.user_id = :userId`,
+      {
+        type: QueryTypes.SELECT,
+        replacements: {
+          userId
+        }
+      }
+    );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The query to fetch user topics is not parameterized, which can lead to SQL injection attacks. Always use parameterized queries or prepared statements to prevent this.

```diff
-    let user_topics = await sequelize.query(
-      `select tp.topic_id from topics as tp
-      left join user_topics as utp on tp.topic_id = utp.topic_id
-      where utp.user_id = :userId`,
-      {
-        type: QueryTypes.SELECT,
-        replacements: {
-          userId
-        }
-      }
-    );
+    let user_topics = await sequelize.models.user_topic.findAll({
+      where: { user_id: userId },
+      attributes: ['topic_id']
+    });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->